### ISSUE TITLE
Unable to connect to urls through domains

### ISSUE THREAD

**Issue by KlutzyBubbles** [kurokobo/awx-on-k3s#178]
## Environment

- OS: Ubuntu 20.04.4
- Kubernetes/K3s: 1.22.7+k3sl
- AWX Operator: 1.1.3 (Recently upgraded from 0.x.x)

## Description

Unable to make requests to urls with domains (trying to sync projects through git).

getting `Failed to connect to git.example.com port 80: Connection timed out`
git.example.com is not actually that, its just changed. Locally there is a dns server that resolves it to the correct ip.

i have also tried github.com repo with the same error

If i use the local ip instead of the domain name then it connects just fine but errors after when it tries to query ansible galaxy with this:

`'https://galaxy.ansible.com/api/': <urlopen error timed out>`

## Step to Reproduce

Dont think there are any steps to reproduce, i am just looking for some help to try debug this instead of having to re install

## Logs

I have also tried nsenter to (what i believe i was doing) lokup the dns entry from the container

```bash
sudo nsenter -t 4550 -n dig git.example.com @10.43.0.10

; <<>> DiG 9.16.1-Ubuntu <<>> git.example.com @10.43.0.10
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 30274
;; flags: qr aa rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
; COOKIE: 15278d6cfcbad4ef (echoed)
;; QUESTION SECTION:
;git.example.com.          IN      A

;; ANSWER SECTION:
git.example.com.   30      IN      A       192.168.5.15

;; Query time: 0 msec
;; SERVER: 10.43.0.10#53(10.43.0.10)
;; WHEN: Wed Jan 25 08:25:32 UTC 2023
;; MSG SIZE  rcvd: 97

```

Not sure if its related but my pods look odd ever since i checked while updating

```bash
klutzybubbles@awx:~/awx-operator$ kubectl -n awx get all
NAME                                                   READY   STATUS                   RESTARTS        AGE
pod/awx-6c645b554-h5zm5                                0/4     Error                    104             315d
pod/awx-6c645b554-mc5xd                                0/4     ContainerStatusUnknown   4 (217d ago)    223d
pod/awx-6c645b554-8cjpt                                0/4     Error                    4               217d
pod/awx-6c645b554-x2c8x                                0/4     Error                    8               204d
pod/awx-6c645b554-v9zpq                                0/4     Error                    8               192d
pod/awx-6c645b554-f29dj                                0/4     Error                    4               186d
pod/awx-6c645b554-whcxh                                0/4     Error                    28              181d
pod/awx-6c645b554-24z45                                0/4     Error                    8               172d
pod/awx-6c645b554-qgrz8                                0/4     Error                    8               165d
pod/awx-6c645b554-vj8m4                                0/4     Error                    12              152d
pod/awx-6c645b554-dql9v                                0/4     Error                    24              138d
pod/awx-6c645b554-vwdpf                                0/4     ContainerStatusUnknown   8 (5d13h ago)   95d
pod/awx-postgres-13-0                                  1/1     Running                  1 (9h ago)      26h
pod/awx-operator-controller-manager-5cbfb95968-tx4nb   2/2     Running                  6 (9h ago)      46h
pod/awx-5c8659c488-28s9s                               4/4     Running                  4 (9h ago)      26h

NAME                                                      TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
service/awx-operator-controller-manager-metrics-service   ClusterIP   10.43.165.111   <none>        8443/TCP   315d
service/awx-postgres-13                                   ClusterIP   None            <none>        5432/TCP   46h
service/awx-service                                       ClusterIP   10.43.158.177   <none>        80/TCP     315d

NAME                                              READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/awx-operator-controller-manager   1/1     1            1           315d
deployment.apps/awx                               1/1     1            1           315d

NAME                                                         DESIRED   CURRENT   READY   AGE
replicaset.apps/awx-operator-controller-manager-5ddf49cc4f   0         0         0       315d
replicaset.apps/awx-6c645b554                                0         0         0       315d
replicaset.apps/awx-operator-controller-manager-5cbfb95968   1         1         1       46h
replicaset.apps/awx-5c8659c488                               1         1         1       46h

NAME                               READY   AGE
statefulset.apps/awx-postgres-13   1/1     46h

```

Sorry if i am not providing enough information, i am fairly new to kubernetes, this was only meant to be a setup so i could learn / test awx but awx required kuberentes and the default minikube option didnt work.

Please let me know what information i should include if i have (most definitely) missed anything.

**Comment by kurokobo** [comment]
@KlutzyBubbles 
To determine if it is a dns-related issue, ensure domain names can be resolved inside the pod. Could you please share your output of following commands?

```bash
kubectl -n kube-system get pods -o wide

kubectl run -it --rm --restart=Never busybox --image=busybox:1.28 -- cat /etc/resolv.conf
kubectl run -it --rm --restart=Never busybox --image=busybox:1.28 -- wget https://galaxy.ansible.com/

kubectl run -it --rm --restart=Never busybox --image=busybox:1.28 -- nslookup git.example.com
kubectl run -it --rm --restart=Never busybox --image=busybox:1.28 -- nslookup galaxy.ansible.com
```

If the domain names can be resolved, I guess your AWX has been placed behind the HTTP proxy.
You can add `HTTP_PROXY`, `HTTPS_PROXY`, `NO_PROXY` through AWX's GUI. Refer my guide about proxy: https://github.com/kurokobo/awx-on-k3s/blob/main/tips/use-http-proxy.md#add-proxy-settings-to-awx

**Comment by KlutzyBubbles** [comment]
> @KlutzyBubbles To determine if it is a dns-related issue, ensure domain names can be resolved inside the pod. Could you please share your output of following commands?
> 
> ```shell
> kubectl -n kube-system get pods -o wide
> 
> kubectl run -it --rm --restart=Never busybox --image=busybox:1.28 -- cat /etc/resolv.conf
> kubectl run -it --rm --restart=Never busybox --image=busybox:1.28 -- wget https://galaxy.ansible.com/
> 
> kubectl run -it --rm --restart=Never busybox --image=busybox:1.28 -- nslookup git.example.com
> kubectl run -it --rm --restart=Never busybox --image=busybox:1.28 -- nslookup galaxy.ansible.com
> ```
> 
> If the domain names can be resolved, I guess your AWX has been placed behind the HTTP proxy. You can add `HTTP_PROXY`, `HTTPS_PROXY`, `NO_PROXY` through AWX's GUI. Refer my guide about proxy: https://github.com/kurokobo/awx-on-k3s/blob/main/tips/use-http-proxy.md#add-proxy-settings-to-awx
> 
> If the domain names cannot be resolved, there might be dns-related issue. Try restaring CoreDNS pod first,
> 
> ```shell
> kubectl -n kube-system delete pod -l k8s-app=kube-dns
> ```
> 
> If the issue still exists, you can explicitly specify dns server to be used by following step 3 to 6 of this guide: https://github.com/kurokobo/awx-on-k3s/blob/main/tips/expose-hosts.md

TLDR; https://github.com/kurokobo/awx-on-k3s/blob/main/tips/expose-hosts.md steps 3-6 solved the issue.

But for anyone that may find this in searching in the future, I had initially thought my DNS servers were setup fine because i had already used the busybox image to make requests with no issues. However for whatever reason explicitly specifying the dns server fixed it.

Not sure if there is any point, thought i should include it anyway, i am still a bit confused on how the outputs from the commands stayed the same even after the dns change

```bash
klutzybubbles@awx:~$ kubectl -n kube-system get pods -o wide
NAME                                      READY   STATUS      RESTARTS         AGE    IP            NODE                   NOMINATED NODE   READINESS GATES
helm-install-traefik-crd--1-9wm78         0/1     Completed   0                316d   10.42.0.4     awx.klutzybubbles.me   <none>           <none>
helm-install-traefik--1-wxm74             0/1     Completed   2                316d   10.42.0.5     awx.klutzybubbles.me   <none>           <none>
local-path-provisioner-84bb864455-q6pwv   1/1     Running     62 (121m ago)    316d   10.42.0.221   awx.klutzybubbles.me   <none>           <none>
svclb-traefik-n2z5b                       2/2     Running     131 (121m ago)   316d   10.42.0.215   awx.klutzybubbles.me   <none>           <none>
coredns-96cc4f57d-dkjnl                   1/1     Running     62 (121m ago)    316d   10.42.0.218   awx.klutzybubbles.me   <none>           <none>
traefik-56c4b88c4b-zkrs4                  1/1     Running     62 (121m ago)    316d   10.42.0.222   awx.klutzybubbles.me   <none>           <none>
metrics-server-ff9dbcb6c-tq6pd            1/1     Running     62 (121m ago)    316d   10.42.0.216   awx.klutzybubbles.me   <none>           <none>
klutzybubbles@awx:~$ kubectl run -it --rm --restart=Never busybox --image=busybox:1.28 -- cat /etc/resolv.conf
search default.svc.cluster.local svc.cluster.local cluster.local klutzybubbles.me
nameserver 10.43.0.10
options ndots:5
pod "busybox" deleted
klutzybubbles@awx:~$ kubectl run -it --rm --restart=Never busybox --image=busybox:1.28 -- wget https://galaxy.ansible.com/
Connecting to galaxy.ansible.com (172.67.68.251:443)
index.html           100% |*******************************|  2275   0:00:00 ETA
pod "busybox" deleted
klutzybubbles@awx:~$ kubectl run -it --rm --restart=Never busybox --image=busybox:1.28 -- nslookup git.klutzybubbles.me
Server:    10.43.0.10
Address 1: 10.43.0.10 kube-dns.kube-system.svc.cluster.local

Name:      git.klutzybubbles.me
Address 1: 192.168.5.15 git.klutzybubbles.me
pod "busybox" deleted
klutzybubbles@awx:~$ kubectl run -it --rm --restart=Never busybox --image=busybox:1.28 -- nslookup galaxy.ansible.com
If you dont see a command prompt, try pressing enter.
Address 1: 2606:4700:20::681a:ea
Address 2: 2606:4700:20::681a:1ea
Address 3: 2606:4700:20::ac43:44fb
Address 4: 104.26.1.234
Address 5: 104.26.0.234
Address 6: 172.67.68.251
pod "busybox" deleted
klutzybubbles@awxkubectl -n kube-system delete pod -l k8s-app=kube-dns
pod "coredns-96cc4f57d-r9rp6" deleted
klutzybubbles@awx:~$ kubectl -n kube-system get pods -o wide
NAME                                      READY   STATUS      RESTARTS         AGE    IP            NODE                   NOMINATED NODE   READINESS GATES
local-path-provisioner-84bb864455-q6pwv   1/1     Running     62 (3h1m ago)    316d   10.42.0.221   awx.klutzybubbles.me   <none>           <none>
svclb-traefik-n2z5b                       2/2     Running     131 (3h1m ago)   316d   10.42.0.215   awx.klutzybubbles.me   <none>           <none>
traefik-56c4b88c4b-zkrs4                  1/1     Running     62 (3h1m ago)    316d   10.42.0.222   awx.klutzybubbles.me   <none>           <none>
metrics-server-ff9dbcb6c-tq6pd            1/1     Running     62 (3h1m ago)    316d   10.42.0.216   awx.klutzybubbles.me   <none>           <none>
helm-install-traefik-crd--1-wqxd8         0/1     Completed   0                15m    10.42.0.231   awx.klutzybubbles.me   <none>           <none>
helm-install-traefik--1-p92sr             0/1     Completed   0                15m    10.42.0.230   awx.klutzybubbles.me   <none>           <none>
coredns-96cc4f57d-rr2bh                   1/1     Running     0                15m    10.42.0.232   awx.klutzybubbles.me   <none>           <none>

```


But like i said the issue has been solved, thank you for the help :)
